crates/aranya-daemon/src/api.rs

```diff
@@ -567,6 +608,153 @@ impl DaemonApi for DaemonApiHandler {
         }
         Err(anyhow!("unable to find AfcBidiChannelReceived effect").into())
     }
+    /// Query devices on team.
+    #[instrument(skip(self))]
+    async fn query_devices_on_team(
+        self,
+        _: context::Context,
+        team: TeamId,
+    ) -> ApiResult<Vec<DeviceId>> {
+        let (_ctrl, effects) = self
+            .client
+            .actions(&team.into_id().into())
+            .query_devices_on_team_off_graph()
+            .await?;
+        let mut devices: Vec<DeviceId> = Vec::new();
+        for e in effects {
+            if let Effect::QueryDevicesOnTeamResult(e) = e {
+                devices.push(e.user_id.into());
+            }
+        }
+        return Ok(devices);
+    }
+    /// Query device role.
+    #[instrument(skip(self))]
+    async fn query_device_role(
+        self,
+        _: context::Context,
+        team: TeamId,
+        device: DeviceId,
+    ) -> ApiResult<ApiRole> {
+        let (_ctrl, effects) = self
+            .client
+            .actions(&team.into_id().into())
+            .query_device_role_off_graph(device.into_id().into())
+            .await?;
+        if let Some(Effect::QueryDeviceRoleResult(e)) =
+            find_effect!(&effects, Effect::QueryDeviceRoleResult(_e))
+        {
+            return Ok(ApiRole::from(e.role));
+        };
+        Err(anyhow!("unable to query device role").into())
+    }
+    /// Query device keybundle.
+    #[instrument(skip(self))]
+    async fn query_device_keybundle(
+        self,
+        _: context::Context,
+        team: TeamId,
+        device: DeviceId,
+    ) -> ApiResult<ApiKeyBundle> {
+        let (_ctrl, effects) = self
+            .client
+            .actions(&team.into_id().into())
+            .query_device_keybundle_off_graph(device.into_id().into())
+            .await?;
+        if let Some(Effect::QueryDeviceKeyBundleResult(e)) =
+            find_effect!(&effects, Effect::QueryDeviceKeyBundleResult(_e))
+        {
+            return Ok(ApiKeyBundle::from(e.user_keys.clone()));
```

Comment:
Nit: The key bundle doesn't need to be cloned

```suggestion
            find_effect!(effects, Effect::QueryDeviceKeyBundleResult(_e))
        {
            return Ok(ApiKeyBundle::from(e.user_keys));
```

Reply:
If I remove `clone()`:
```
error[E0507]: cannot move out of `e.user_keys` which is behind a shared reference
   --> crates/aranya-daemon/src/api.rs:667:42
    |
667 |             return Ok(ApiKeyBundle::from(e.user_keys));
    |                                          ^^^^^^^^^^^ move occurs because `e.user_keys` has type `policy::KeyBundle`, which does not implement the `Copy` trait
```

Reply:
You can pass the effects variable by value rather than using a reference. It doesn't look like it's used after line 665